lib/gauge_qcharge.cu

```diff
@@ -0,0 +1,120 @@
+#include <quda_internal.h>
+#include <tune_quda.h>
+#include <gauge_field.h>
+
+#define DOUBLE_TOL 1e-15
+#define SINGLE_TOL 2e-6
+
+#include <jitify_helper.cuh>
+#include <kernels/gauge_qcharge.cuh>
+
+namespace quda
+{
+
+#ifdef GPU_GAUGE_TOOLS
+
+  template <typename Float, typename Arg> class QChargeCompute : TunableLocalParity
+  {
+
+    Arg &arg;
+    const GaugeField &meta;
+
+private:
+    bool tuneGridDim() const { return true; }
+    unsigned int minThreads() const { return arg.threads; }
+
+public:
+    QChargeCompute(Arg &arg, const GaugeField &meta) : arg(arg), meta(meta) {}
+#ifdef JITIFY
+    create_jitify_program("kernels/gauge_qcharge.cuh");
+#endif
+    virtual ~QChargeCompute() {}
+
+    void apply(const cudaStream_t &stream)
+    {
+      if (meta.Location() == QUDA_CUDA_FIELD_LOCATION) {
+        arg.result_h[0] = 0.;
+        TuneParam tp = tuneLaunch(*this, getTuning(), getVerbosity());
+#ifdef JITIFY
+        using namespace jitify::reflection;
+        jitify_error = program->kernel("quda::qChargeComputeKernel")
+                         .instantiate(Type<Float>(), Type<Arg>())
+                         .configure(tp.grid, tp.block, tp.shared_bytes, stream)
+                         .launch(arg);
+#else
+        // DMH: Hardcoded 32 blockSize for reduction kernel
```

Comment:
Curious: why not keep the prior `LAUNCH_KERNEL` here which instantiates all block sizes.

Reply:
Relic from an attempt create a function that returned individual q(x) values. Thanks for the catch.